### PR TITLE
Feature/issue/65/tag supportt to client and spec

### PR DESCRIPTION
**Pull request by gabrielbussolo**
# Summary of this proposal

add tags to API spec and to client

## Issue(s) number(s)

*Disclaimer: Please do not create a Pull Request without creating an issue first.*

## Checklist for submitter

- [ ] My PR has a related issue/bug number.
- [ ] My PR provides tests
  - [ ] Integration tests are added/passes
  - [ ] Unit tests are added/passes
  - [ ] End-to-end tests added/passes
  - [ ] Static code check added/passes
  - [ ] Linting on documentation added/passes
  - [ ] Does not affect code coverage stats
- [ ] My PR requires updating dependencies
- [ ] My PR has the documentation changes required.

## Checklist for reviewer

- [ ] The proposal fixes a bug/issue or implements a new feature that is well described.
- [ ] The proposal has sufficient test cases that covers the changes.
  - [ ] If changes an API, it does not break backwards compatibility (-1 major version).
- [ ] If integration is required, the proposal has integration tests.
- [ ] The proposal does not break coverage stats
- [ ] The proposal has the required documentation changes.

## Backport

- [ ] Backport to the latest stable release.

### CODE REVIEW

File: spec/open-api.yml [hunk 1]
```diff
@@ -2291,6 +2319,11 @@ paths:
           in: query
           name: name
           description: Name matching pipelines.
+        - schema:
+            type: string
+          in: query
+          name: tags_query
+          description: Tag query.
```
Comment (niedbalski) [inline spec/open-api.yml:2326]: please add an example query.

File: spec/open-api.yml [hunk 2]
```diff
@@ -2341,6 +2374,11 @@ paths:
...
```
Comment (niedbalski) [inline spec/open-api.yml:2381]: please add an example query.

File: spec/open-api.yml [hunk 3]
```diff
@@ -2044,6 +2067,11 @@ paths:
           in: query
           name: name
           description: Name matching aggregators.
+        - schema:
+            type: string
+          in: query
+          name: tags_query
```
Comment (niedbalski) [inline spec/open-api.yml:2073]: please add an example query.

**Review by niedbalski — CHANGES_REQUESTED** [review]
minor nitpicks on the schema.

File: client/agent_test.go [hunk 4]
```diff
@@ -117,6 +117,50 @@ func TestClient_Agents(t *testing.T) {
...
+		wantEqual(t, len(tag2.Items), 5)
+		wantEqual(t, tag2.Items[0].Tags, []string{"tagtwo", "tagthree"})
+
+		opts.TagFilter("tagthree")
+		tag3, err := asUser.Agents(ctx, project.ID, opts)
+		wantEqual(t, err, nil)
+		wantEqual(t, len(tag3.Items), 10)
+		wantEqual(t, tag3.Items[0].Tags, []string{"tagone", "tagthree"})
+
+		opts.TagFilter("tagone AND tagthree")
```
Comment (nicolasparada) [inline client/agent_test.go:158]: A better test would be `tagone AND tagtwo` and expect 0 items.

File: types/agent.go [hunk 5]
```diff
@@ -81,6 +81,10 @@ type AgentsParams struct {
 	Tags   *string
 }
 
+func (p *AgentsParams) TagFilter(s string) {
```
Comment (nicolasparada) [inline types/agent.go:84]: I don't know about these helpers 🤔 
We are not doing it like thise anywhere else.

I'm not against it, but also we should be consistent.

Comment (niedbalski) [reply types/agent.go:84]: I don't think we need them, as we have parameters already for this. Maybe in the future when we compose more complex tag queries

Comment (gabrielbussolo) [reply types/agent.go:84]: i got from the issue example, should i remove so? 🤔

**Review by nicolasparada — APPROVED** [review]
LGTM 👍

**Pull request merged**